bitcoin/src/address.rs

```diff
@@ -434,13 +434,13 @@ impl Payload {
     /// Constructs a [Payload] from an output script (`scriptPubkey`).
     pub fn from_script(script: &script::Script) -> Result<Payload, Error> {
         Ok(if script.is_p2pkh() {
-            let mut hash_inner = [0u8; 20];
-            hash_inner.copy_from_slice(&script.as_bytes()[3..23]);
-            Payload::PubkeyHash(PubkeyHash::from_inner(hash_inner))
+            let mut bytes = [0u8; 20];
+            bytes.copy_from_slice(&script.as_bytes()[3..23]);
+            Payload::PubkeyHash(PubkeyHash::from_byte_array(bytes))
```

Comment:
I believe this is simpler:

```suggestion
            let bytes = script.as_bytes()[3..23].try_into().expect("statically 20B long");
            Payload::PubkeyHash(PubkeyHash::from_byte_array(bytes))
```

Reply:
Used suggestion, as well as below. Thanks